### PR TITLE
Port fastdds-discovery-server to DDS API [13782]

### PR DESCRIPTION
**Pull request by VictorMMaroto**

### CODE REVIEW

File: tools/fds/server.cpp [hunk 1]
```diff
@@ -269,10 +270,8 @@ int main (
         cout << endl << "### Server shut down ###" << endl;
```
Comment (MiguelCompany) [inline tools/fds/server.cpp:270]: I would delete the participant here:

```suggestion
        cout << endl << "### Server shut down ###" << endl;
        DomainParticipantFactory::get_instance()->delete_participant(pServer);
```

Comment (VictorMMaroto) [reply tools/fds/server.cpp:270]: added

**Review by MiguelCompany — APPROVED** [review]
LGTM

File: tools/fds/server.cpp [hunk 2]
```diff
@@ -230,7 +229,8 @@ int main (
 
     // Create the server
     int return_value = 0;
-    Participant* pServer = Domain::createParticipant(*att, nullptr);
+    // Participant* pServer = Domain::createParticipant(*att, nullptr);
```
Comment (EduPonz) [inline tools/fds/server.cpp:232]:
```suggestion
```

File: tools/fds/server.cpp [hunk 3]
```diff
@@ -26,10 +26,11 @@
...
-#include <fastrtps/Domain.h>
-#include <fastrtps/participant/Participant.h>
+#include <fastdds/dds/domain/DomainParticipant.hpp>
+#include <fastdds/dds/domain/DomainParticipantFactory.hpp>
 #include <fastdds/dds/log/Log.hpp>
 
+using namespace eprosima::fastdds::dds;
 using namespace eprosima;
 using namespace fastrtps;
 using namespace std;
```
Comment (EduPonz) [inline tools/fds/server.cpp:36]: Do we need these?

**Review by EduPonz — CHANGES_REQUESTED** [review]
Nitpicks

File: tools/fds/server.cpp (hunk 3 above)
Comment (VictorMMaroto) [reply tools/fds/server.cpp:36]: Yes, DomainParticipantQos in dds namespace

**Comment by EduPonz** [comment]
@richiprosima please test this

**Review by MiguelCompany — DISMISSED** [review]
LGTM with green CI

**Review by EduPonz — APPROVED** [review]

**Pull request merged**